tls/s2n_handshake_io.c

```diff
@@ -1386,8 +1401,14 @@ static S2N_RESULT s2n_handshake_app_data_recv(struct s2n_connection *conn)
  */
 static int s2n_handshake_read_io(struct s2n_connection *conn)
 {
-    uint8_t record_type;
-    uint8_t message_type;
+    POSIX_ENSURE_REF(conn);
+
+    /* Ensure the state machine referenced is consistent throughout the handshake */
+    if (conn->handshake.state_machine_fixed) {
+        POSIX_ENSURE(IS_TLS13_HANDSHAKE(conn) == conn->handshake.tls13_state_machine, S2N_ERR_STATE_MACHINE_SWITCH);
+    }
+
+    uint8_t record_type, message_type = 0;
```

Comment:
```suggestion

    uint8_t record_type = 0, message_type = 0;
```
Or you might just want to declare these on separate lines for clarity, if the C init behavior isn't obvious.